zerver/lib/actions.py

```diff
@@ -4260,9 +4260,16 @@ def do_update_message(user_profile: UserProfile, message: Message, topic_name: O
     }  # type: Dict[str, Any]
     changed_messages = [message]
 
+    users_to_be_notified_ids = set()  # type: Set[int]
     if message.is_stream_message():
         stream_id = message.recipient.type_id
-        event['stream_name'] = Stream.objects.get(id=stream_id).name
+        thread_being_edited = Stream.objects.get(id=stream_id)
+        event['stream_name'] = thread_being_edited.name
+
+        if thread_being_edited.is_history_public_to_subscribers:
+            subscribers = get_active_subscriptions_for_stream_id(stream_id)
+            # All users that are subscribed to the stream must be notified when a message is edited
+            users_to_be_notified_ids = set([user.user_profile_id for user in subscribers])
```

Comment:
Wouldn't it be more readable to move some of this code to the segment of code much lower, where the `` send_event `` stuff happens? As it is now, `` users_to_be_notified_ids `` gets defined here and not used for a big chunk of code, until near the end of this function. And at that point reader probably no longer remembers when or how it was defined.

Reply:
Yeah, it was near the send_event call before. I put it there because otherwise, I would repeat the check `if message.is_stream_message()`, but it really seems better to repeat it and improve readability